allennlp/training/trainer.py

```diff
@@ -226,17 +243,17 @@ def on_batch(
 
         if not is_primary:
             return None
-        assert self._tensorboard is not None
+        assert self._log_writer is not None
 
-        if self._tensorboard.should_log_histograms_this_batch():
+        if self._log_writer.should_log_distributions_this_batch():
             assert self._param_updates is not None
             for name, param in trainer.model.named_parameters():
                 self._param_updates[name].sub_(param.detach().cpu())
         else:
             self._param_updates = None
 
-        self._tensorboard.log_memory_usage(cpu_memory_usage, gpu_memory_usage)
-        self._tensorboard.log_batch(
+        self._log_writer.log_memory_usage(cpu_memory_usage, gpu_memory_usage)
```

Comment:
I think this was my fault, but we might as well fix it in this PR. We shouldn't log mem and GPU usage on every batch because that's rather costly. We could either remove this entirely (as well as lines 241 and 242 above) or hide it behind a flag. I'm in favor of removing entirely.